src/vs/platform/actions/common/actions.ts

```diff
@@ -33,6 +33,7 @@ export interface ISubmenuItem {
 	order?: number;
 	isSelection?: boolean;
 	rememberDefaultAction?: boolean;	// for dropdown menu: if true the last executed action is remembered as the default action
+	toggled?: ContextKeyExpression;
```

Comment:
I doubt that a submenu should be toggled. What should that mean in the domain of actions, menus and submenus? Submenus are just containers for other menus. I do understand the use-case that's tackled here but I believe that's UI only. E.g, the rendering of a submenu should check its actions and render accordingly but it shouldn't have a toggled properties itself.

Reply:
e.g [SubmenuEntryActionViewItem](https://github.com/microsoft/vscode/blob/fa45b00051606aa4e7bd8b4da05f209faa5ba02d/src/vs/platform/actions/browser/menuEntryActionViewItem.ts#L295) is IMO the better place to implement this, without new model properties/concept but purely rendering based